Automattic/pocket-casts-android
# [Paywall Experiment] Tweaks for Upgrade Bottom Sheet

## Description
- Updates the existing bottom sheet to match with SEQe6JBDCeegcpa9Bfg5Ww-fi-3_756
- Updates title and button text for Plus subscription.  p1724156035089779/1724086549.470929-slack-C07G1MH890E
- Updates Pill format
- Apply gradient background effect

Fixes #2676

## Testing Instructions
1. Run the app in debug
2. Have PAYWALL_AB_EXPERIMENT feature flag enabled
3. Log in with free user account
4. Go to Podcasts tab
5. Tap on the folder at the top to open the paywall
6. Tap on `Get Pocket Casts Plus` button to open the bottom sheet
7. Verify the changes described in description section were applied  ✅


## Screenshots or Screencast 

| Before | After |
|--------|--------|
| <img src="https://github.com/user-attachments/assets/ac51ea56-c5bd-46e0-8fbc-6353d067fbe3" width="400px"> | <img src="https://github.com/user-attachments/assets/fe406a31-39a5-4005-9eea-956b275c54db"  width="400px"> |


## Checklist
 
#### I have tested any UI changes...
